codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketProtocolHandler.java

```diff
@@ -82,29 +83,29 @@ private static void readIfNeeded(ChannelHandlerContext ctx) {
     }
 
     @Override
-    public void close(final ChannelHandlerContext ctx, final ChannelPromise promise) throws Exception {
+    public void close(final ChannelHandlerContext ctx, final ChannelOutboundInvokerCallback callback) throws Exception {
         if (closeStatus == null || !ctx.channel().isActive()) {
-            ctx.close(promise);
+            ctx.close(callback);
         } else {
             if (closeSent == null) {
                 write(ctx, new CloseWebSocketFrame(closeStatus), ctx.newPromise());
             }
-            flush(ctx);
+            flush(ctx, );
```

Comment:
fyi

Reply:
yep.... Like I said only focus on `common` , `transport`, `codec` for now. This is still a WIP.